### PR TITLE
{math}[foss/2018b] ArrayFire v3.6.4

### PR DESCRIPTION
**Pull request by JackPerdue**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by JackPerdue** [comment]
What am I supposed to do for checksums when doing recursive git pulls?

**Comment by JackPerdue** [comment]
(note: it DID pass --check-style)

**Comment by JackPerdue** [comment]
ArrayFire doesn't seem to have any MPI support so I didn't need a CUDA-enabled MPI.  It was easier just to add CUDA to the foss toolchain (instead of building all needed components with fosscuda).

My concern about checksums was that since this uses a recursive git pull that the checksum might vary from pull to pull.  I'm guessing now that the use of a certain git "tag" guarantees getting the same files (both main component and subcomponents).  Anyway... added.

**Comment by smoors** [comment]
@JackPerdue ah I get what you mean now.
as explained here:
https://easybuild.readthedocs.io/en/latest/Writing_easyconfig_files.html#common-easyconfig-param-sources-git-config
the checksum is not guaranteed to be the same across different systems.
maybe it's better to use the release tarball that contains all sub-modules here: http://arrayfire.com/arrayfire_source/arrayfire-full-3.6.4.tar.bz2

**Comment by smoors** [comment]
@JackPerdue you also have to use `CUDA-9.2.88` since that is the version that is used as a dep in other easyconfigs.

**Comment by JackPerdue** [comment]
OK... new CUDA build shortly with aligned version (the CUDA version is an hour+ build).

Where did you find that download URL/link???  I'd prefer, as would EB given the link above, to just download a full source release.   After I dug around I found their archives with this link:  http://arrayfire.s3.amazonaws.com/arrayfire_source/arrayfire-full-3.6.4.tar.bz2  which is a different URL than your link hosted at  the ArrayFire site.  Was your link mentioned on their site somewhere? (and if so, where?)

**Comment by JackPerdue** [comment]
@smoors: I guess I'll use your URL in case they move from Amazon and redirect somewhere else.

**Comment by smoors** [comment]
@JackPerdue I found the link in their releases tab here: https://github.com/arrayfire/arrayfire/releases

**Comment by JackPerdue** [comment]
Oh... I overlooked that (I typically just see the links just below the description.  Thanks.  CUDA build with non-git source in progress (seems to take much longer here than on Travis).

File: easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b-CUDA-9.2.88.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Authors:: Jack Perdue <j-perdue@tamu.edu> - TAMU HPRC - http://hprc.tamu.edu
+
+easyblock = 'CMakeMake'
+
+name = 'ArrayFire'
+version = '3.6.4'
+cudaver = '9.2.88'
+versionsuffix = '-CUDA-%s' % cudaver
+gittag = 'v%s' % version
```
Comment (yoniv1) [inline easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b-CUDA-9.2.88.eb]: gittag is not used anymore, can be removed.

File: easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Authors:: Jack Perdue <j-perdue@tamu.edu> - TAMU HPRC - http://hprc.tamu.edu
+
+easyblock = 'CMakeMake'
+
+name = 'ArrayFire'
+version = '3.6.4'
+gittag = 'v%s' % version
```
Comment (yoniv1) [inline easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b.eb]: gittag is not used anymore, can be removed.

File: easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b.eb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+]
+
+dependencies = [
+    ('Boost', '1.67.0'),
+]
+
+separate_build_dir = True
+
+sanity_check_paths = {
+    'files': ['include/af/version.h', 'lib64/libaf.so'],
```
Comment (yoniv1) [inline easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b.eb]: please replace with this:
```
'files': ['include/af/version.h', 'lib64/libaf.%s' % SHLIB_EXT],
```

File: easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b-CUDA-9.2.88.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    ('Boost', '1.67.0'),
+    ('CUDA', cudaver, '', True),
+]
+
+separate_build_dir = True
+
+configopts = '-DAF_BUILD_OPENCL=OFF'
+
+sanity_check_paths = {
+    'files': ['include/af/version.h', 'lib64/libaf.so'],
```
Comment (yoniv1) [inline easybuild/easyconfigs/a/ArrayFire/ArrayFire-3.6.4-foss-2018b-CUDA-9.2.88.eb]: please replace with this:
```
'files': ['include/af/version.h', 'lib64/libaf.%s' % SHLIB_EXT],
```

**Comment by smoors** [comment]
@yoniv1 sorry that was me, forgot to log out from your account...

**Comment by JackPerdue** [comment]
Travis glitch?  Or did I break something?

**Comment by JackPerdue** [comment]
Nevermind... I missed the SHLIB_EXT comments earlier and will need to resubmit anyway.

**Comment by JackPerdue** [comment]
Dunno.... seems to me that Travis is having problems setting up its Python/2.6 enviironment........

**Comment by boegel** [comment]
@JackPerdue Travis issues fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/8483

**Comment by JackPerdue** [comment]
hmm.... OK.... if you say so (who am I to disagree)

**Comment by boegel** [comment]
(close/re-open to make Travis CI happy w.r.t. Python 2.6)

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
nic423 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/4ed5755eed13a7f11be913056f37fcc2 for a full test report.

**Comment by smoors** [comment]
Going in, thanks @JackPerdue!

**Pull request merged**